test/unit/math/prim/fun/svd_U_test.cpp

```diff
@@ -0,0 +1,14 @@
+#include <stan/math/prim.hpp>
+#include <gtest/gtest.h>
+#include <test/unit/util.hpp>
+
+TEST(MathMatrixPrimMat, svd_U) {
```

Comment:
For the tests in prim, test a 0x0 matrix, a 1x1 matrix, a small square matrix (2x2), a small tall rectangular matrix (3x2), and a small wide rectangular matrix (2x3).

We need to be sure the prim function is working before we can trust the `expect_ad` stuff to work (since it depends on finite differences as reference)